plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/api/tools/CreateEdgeUnsyncBugTest.java

```diff
@@ -0,0 +1,262 @@
+ *******************************************************************************/
+package org.eclipse.sirius.tests.unit.api.tools;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.sirius.business.api.preferences.SiriusPreferencesKeys;
+import org.eclipse.sirius.diagram.DDiagram;
+import org.eclipse.sirius.diagram.DEdge;
+import org.eclipse.sirius.diagram.EdgeTarget;
+import org.eclipse.sirius.tests.SiriusTestsPlugin;
+import org.eclipse.sirius.tests.support.api.SiriusDiagramTestCase;
+import org.eclipse.sirius.tests.support.api.TestsUtil;
+import org.eclipse.sirius.ui.business.api.dialect.DialectUIManager;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.DRepresentationElement;
+import org.eclipse.ui.IEditorPart;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import junit.framework.AssertionFailedError;
+
+/**
+ * Test the creation of edges
+ * 
+ * @author Séraphin Costa
+ * 
+ */
+@RunWith(value = Parameterized.class)
+public class CreateEdgeUnsyncBugTest extends SiriusDiagramTestCase {
+
+    private static final String PATH = "/" + SiriusTestsPlugin.PLUGIN_ID + "/data/unit/tool/bugEdge/";
+
+    private static final String SEMANTIC_MODEL_PATH = PATH + "My.ecore";
+
+    private static final String REPRESENTATION_MODEL_PATH = PATH + "representations.aird";
+
+    private static final String MODELER_PATH = PATH + "project.odesign";
+
+    private static final String VIEWPOINT_NAME = "EdgeTest";
+
+    private DDiagram diagram;
+
+    private IEditorPart editorPart;
+
+    private boolean autoRefresh;
+
+    private boolean unsynchronizedDiagram;
+
+    public CreateEdgeUnsyncBugTest(String name, boolean autoRefresh, boolean unsynchronizedDiagram) {
+        this.autoRefresh = autoRefresh;
+        this.unsynchronizedDiagram = unsynchronizedDiagram;
+    }
+
+    @Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        // {name, autoRefresh, unsynchronizedDiagram}
+        // @formatter:off
+        return Arrays.asList(new Object[][] {
+            { "manual refresh, sync", false, false },
+            { "manual refresh, unsync", false, true },
+            { "auto refresh, sync", true, false },
+            { "auto refresh, unsync", true, true },
+        });
+        // @formatter:on
+    }
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        genericSetUp(SEMANTIC_MODEL_PATH, MODELER_PATH, REPRESENTATION_MODEL_PATH);
+        initViewpoint(VIEWPOINT_NAME);
```

Comment:
This step is not necessary. The Viewpoint is already activated in your representations file.